File: app/src/main/java/com/example/simplehabitassignment/network/response/GetCategoriesAndProgramsResponse.java
package com.example.simplehabitassignment.network.response;

import com.example.simplehabitassignment.data.vos.CategoriesAndProgramsVO;
import com.google.gson.annotations.SerializedName;


import java.util.List;

public class GetCategoriesAndProgramsResponse {

    @SerializedName("code")
    private int code;

    @SerializedName("message")
    private String message;

    @SerializedName("apiVersion")
    private String apiVersion;

    @SerializedName("page")
    private String page;

    @SerializedName("categoriesPrograms")
    private List<CategoriesAndProgramsVO> categoriesProgramsList;


    public int getCode() {
        return code;
    }

    public String getMessage() {
        return message;
    }

    public String getApiVersion() {
        return apiVersion;
    }

    public String getPage() {
        return page;
    }

    public List<CategoriesAndProgramsVO> getCategoriesProgramsList() {
        return categoriesProgramsList;
    }

    public boolean isResponseSuccessful()
    {
        return code==200;
    }
}

File: app/src/main/java/com/example/simplehabitassignment/network/response/GetCategoriesAndProgramsResponse.kt
package com.example.simplehabitassignment.network.response

import com.example.simplehabitassignment.data.vos.CategoriesAndProgramsVO
import com.google.gson.annotations.SerializedName

class GetCategoriesAndProgramsResponse {

    @SerializedName("code")
    val code: Int = 0

    @SerializedName("message")
    val message: String? = null

    @SerializedName("apiVersion")
    val apiVersion: String? = null

    @SerializedName("page")
    val page: String? = null

    @SerializedName("categoriesPrograms")
    val categoriesProgramsList: List<CategoriesAndProgramsVO>? = null

    val isResponseSuccessful: Boolean
        get() = code == 200
}
